cloudconfig/userdatacfg_test.go

```diff
@@ -725,6 +725,56 @@ func (*cloudinitSuite) TestCloudInitConfigure(c *gc.C) {
 	}
 }
 
+func (s *cloudinitSuite) TestCloudInitConfigCloudInitUserData(c *gc.C) {
+	environConfig := minimalModelConfig(c)
+	environConfig, err := environConfig.Apply(map[string]interface{}{
+		config.CloudInitUserDataKey: validCloudInitUserData,
+	})
+	c.Assert(err, jc.ErrorIsNil)
+	instanceCfg := s.createInstanceConfig(c, environConfig)
+	cloudcfg, err := cloudinit.New("precise")
+	c.Assert(err, jc.ErrorIsNil)
+	udata, err := cloudconfig.NewUserdataConfig(instanceCfg, cloudcfg)
+	c.Assert(err, jc.ErrorIsNil)
+	err = udata.Configure()
+	c.Assert(err, jc.ErrorIsNil)
+
+	// Verify the settings against cloudinit-userdata
+	cfgPackages := cloudcfg.Packages()
+	expectedPackages := []string{
```

Comment:
We should be able to craft the cloud init to have some default packages (and also run commands) so that we check that our user specified packages are appended. So all we need to do is call AddPackage() on cloudcfg above etc.

Reply:
there are already default packages there, added a few more check to ensure at end of package list.